openlibrary-wikidata-bot/jobs/sync_author_identifiers_from_wikidata.py

```diff
@@ -0,0 +1,210 @@
+                    if validate_wikidata_key(obj, "P648") and re.fullmatch(r"OL\d+A", obj["value"]["content"])
+                ]
+
+                if len(ol_ids) == 0:
+                    continue
+                wd_id = parsed_wikidata_json["id"]
+
+                if len(ol_ids) > 1:
+                    for ol_id in ol_ids:
+                        author = ol.Author.get(ol_id)
+                        write_error(wd_id, ol_id, author.name, "multiple_openlibrary_authors_for_one_wikidata_row", "ol_id", f'[{",".join([f'"{val}"' for val in ol_ids])}]')
+                    continue
+                ol_id = ol_ids[0]
+
+                author = ol.Author.get(ol_id)
+
+                remote_ids = {}
+
+                # Check for just the identifiers we're interested in
+                for wd_p_value, remote_ids_key in WD_IDENTIFIERS.items():
+
+                    # Skip if this remote ID isn't in the wikidata json at all
+                    if wd_p_value not in wd_json_top_level_fields:
+                        continue
+                    wd_remote_id_values = wd_json_top_level_fields.get(wd_p_value, [])
+                    if len(wd_remote_id_values) == 0:
+                        continue
+
+                    # Get the value(s) for this remote ID from the wikidata json
+                    valid_wd_remote_id_values = [
+                        obj["value"]["content"]
+                        for obj in wd_remote_id_values
+                        if validate_wikidata_key(obj, wd_p_value)
+                    ]
+                    if len(valid_wd_remote_id_values) == 0:
+                        continue
+                    
+                    # Simple case: there's only one match. Add it to the remote_ids dict
+                    if len(valid_wd_remote_id_values) == 1:
+                        remote_id_value = valid_wd_remote_id_values[0]
+                        remote_ids[remote_ids_key] = remote_id_value
+                    
+                    # Bad case: there are multiple values for the remote ID
+                    elif len(wd_remote_id_values) > 1:
+                        write_error(wd_id, ol_id, author.name, "multiple_wikidata_remote_ids_for_one_author", remote_ids_key, ",".join([f'"{val}"' for val in valid_wd_remote_id_values]))
+
+                if len(remote_ids.keys()) > 0:
+                    remote_ids, update_count = merge_remote_ids(author, remote_ids, wd_id)
+                    if update_count > 0:
+                        if not dry_run:
+                            pass
+                            # I am not trying this yet! I'm terrified! I made dry-run default to false for this reason 😬
+                            # author.remote_ids = remote_ids
+                            # author.save("[sync_author_identifiers_with_wikidata] add wikidata remote identifiers")
+                        logger.info(f'new remote_ids for {ol_id}: {remote_ids}')
```

Comment:
By including the wikidata id we're now guaranteed for this to always be > 0, so can remove the if + dedent